### PR TITLE
VRP: improving extraction of HW and Version

### PR DESCRIPTION
**Pull request by PipoCanaja**
As discussed in https://github.com/librenms/librenms/pull/9159

This new PR tries to extract as much as possible from SysDescr. If it fails, it uses OIDs (starting by the one that is currently use in existing code).

-> On devices that sysDescr matching the regexp, we'll have a speed improvement (no more snmpget) to fill the value
-> On devices that do not match the regexp, if the device uses the 1st OID, no speed difference compared to today. 
-> On devices that do not match sysDescr, nor the 1st OID, then we may improve the information displayed but at the cost of more OID polled at each poller occurence.

PipoCanaja

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

File: includes/polling/os/vrp.inc.php [hunk 1]
```diff
@@ -1,6 +1,37 @@
...
+if (!empty($matches[1])) {
+    $hardware = "Huawei " . trim($matches[1]);
+} else {
+    // The HW cannot be extracted from sysdescr, let's try OIDs
+    $oidList[] = '.1.3.6.1.4.1.2011.2.33.20.1.1.1.3.0';
+    $oidList[] = '.1.3.6.1.4.1.2011.5.25.188.1.4.0';
+    $oidList[] = '.1.3.6.1.4.1.2011.5.25.183.1.25.1.5.1';
+    $oidList[] = '.1.3.6.1.4.1.2011.5.25.31.6.5.0';
+    foreach ($oidList as $oid) {
+        $hardware_tmp = trim(snmp_get($device, $oid, '-OQv'), '"');
```
Comment (murrant) [inline includes/polling/os/vrp.inc.php]: you don't need to trim quotes.

Comment (PipoCanaja) [reply includes/polling/os/vrp.inc.php]: Cleaned, as well as d_echo statements.

**Comment by jozefrebjak** [comment]
Worked for me.

File: includes/polling/os/vrp.inc.php [hunk 2]
```diff
@@ -1,6 +1,33 @@
...
+
+if (!empty($matches[2])) {
+    $version .= " (" . trim($matches[2]) . ")";
+}
+
+if (!empty($matches[1])) {
+    $hardware = "Huawei " . trim($matches[1]);
+} else {
+    // The HW cannot be extracted from sysdescr, let's try OIDs
+    $oidList[] = '.1.3.6.1.4.1.2011.2.33.20.1.1.1.3.0';
```
Comment (murrant) [inline includes/polling/os/vrp.inc.php]: Don't declare an array like this, do it like this:
```php
oidList = [
    '.1.3.6.1.4.1.2011.2.33.20.1.1.1.3.0',
    '.1.3.6.1.4.1.2011.5.25.188.1.4.0',
    '.1.3.6.1.4.1.2011.5.25.183.1.25.1.5.1',
    '.1.3.6.1.4.1.2011.5.25.31.6.5.0',
];

Comment (murrant) [reply includes/polling/os/vrp.inc.php]: They do indeed:
```
HUAWEI-SRM-MIB::hw8060FrameDesc
HUAWEI-DEVICE-EXT-MIB::hwProductName
HUAWEI-STACK-MIB::hwStackDeviceType
HUAWEI-ENTITY-EXTENT-MIB::hwEntitySystemModel
```

Comment (PipoCanaja) [reply includes/polling/os/vrp.inc.php]: Hi,
I did change the order, and resolve as much OIDs as possible considering the MIB Files we have. It is not as nice as your list, but it is improved. 
I also now take SNMP value first and sysDescr later. The idea is that SNMP is most of the time more detailed than sysDescr. 
Bye

**Comment by murrant** [comment]
You need to fix the whitespace formating issue mentioned in travis output.
Also, test data needs to be updated as you have changed some data.
`./scripts/save-test-data.php -o vrp` (if you don't have snmpsim, you can install it with `pip3 install snmpsim`)

**Comment by PipoCanaja** [comment]
Seems that this time the checks went good.

**Pull request merged**